cli_tools/gce_windows_upgrade/upgrader/workflows.go

```diff
@@ -0,0 +1,216 @@
+	}
+
+	stepStopInstance, err := newStep(w, "stop-instance")
+	if err != nil {
+		return w, err
+	}
+	stepStopInstance.StopInstances = &daisy.StopInstances{
+		Instances: []string{params.InstanceURI},
+	}
+
+	stepDetachNewOSDisk, err := newStep(w, "detach-new-os-disk", stepStopInstance)
+	if err != nil {
+		return w, err
+	}
+	stepDetachNewOSDisk.DetachDisks = &daisy.DetachDisks{
+		{
+			Instance:   params.InstanceURI,
+			DeviceName: fmt.Sprintf("projects/%v/zones/%v/devices/%v", params.project, params.zone, params.osDiskDeviceName),
+		},
+	}
+
+	stepAttachOldOSDisk, err := newStep(w, "attach-old-os-disk", stepDetachNewOSDisk)
+	if err != nil {
+		return w, err
+	}
+	stepAttachOldOSDisk.AttachDisks = &daisy.AttachDisks{
+		{
+			Instance: params.InstanceURI,
+			AttachedDisk: compute.AttachedDisk{
+				Source:     params.osDisk,
+				DeviceName: params.osDiskDeviceName,
+				AutoDelete: params.osDiskAutoDelete,
+				Boot:       true,
+			},
+		},
+	}
+
+	stepDetachInstallMediaDisk, err := newStep(w, "detach-install-media-disk", stepAttachOldOSDisk)
+	if err != nil {
+		return w, err
+	}
+	stepDetachInstallMediaDisk.DetachDisks = &daisy.DetachDisks{
+		{
+			Instance:   params.InstanceURI,
+			DeviceName: fmt.Sprintf("projects/%v/zones/%v/devices/%v", params.project, params.zone, installMediaDiskName),
+		},
+	}
+
+	stepRestoreScript, err := newStep(w, "restore-script", stepDetachInstallMediaDisk)
+	if err != nil {
+		return w, err
+	}
+	stepRestoreScript.UpdateInstancesMetadata = &daisy.UpdateInstancesMetadata{
+		{
+			Instance: params.InstanceURI,
+			Metadata: map[string]string{
+				"windows-startup-script-url": originalStartupScriptURL,
+			},
+		},
+	}
+
+	stepStartInstance, err := newStep(w, "start-instance", stepRestoreScript)
+	if err != nil {
+		return w, err
+	}
+	stepStartInstance.StartInstances = &daisy.StartInstances{
+		Instances: []string{params.InstanceURI},
+	}
+
+	stepDeleteNewOSDisk, err := newStep(w, "delete-new-os-disk", stepStartInstance)
+	if err != nil {
+		return w, err
+	}
+	stepDeleteNewOSDisk.DeleteResources = &daisy.DeleteResources{
+		Disks: []string{
+			fmt.Sprintf("projects/%v/zones/%v/disks/%v", params.project, params.zone, newOSDiskName),
+		},
+	}
+
+	stepDeleteUpgradeDisks, err := newStep(w, "delete-install-media-disk", stepStartInstance)
+	if err != nil {
+		return w, err
+	}
+	stepDeleteUpgradeDisks.DeleteResources = &daisy.DeleteResources{
+		Disks: []string{
+			fmt.Sprintf("projects/%v/zones/%v/disks/%v", params.project, params.zone, newOSDiskName),
+			fmt.Sprintf("projects/%v/zones/%v/disks/%v", params.project, params.zone, installMediaDiskName),
+		},
+	}
+
+	setWorkflowAttributes(w, params)
+	err = daisyutils.RunWorkflowWithCancelSignal(ctx, w)
+	return w, err
+}
+
+func newStep(w *daisy.Workflow, name string, dependencies ...*daisy.Step) (*daisy.Step, error) {
+	s, err := w.NewStep(name)
+	if err != nil {
+		return nil, err
+	}
+
+	err = w.AddDependency(s, dependencies...)
+	return s, err
+}
```

Comment:
This is generic and will probably be reused in other CLI tools if we decide to use separate workflow solution to dynamic Daisy. Consider moving to cli_tools/common/utils

Reply:
done